PQAnalysis/io/traj_file/trajectory_reader.py

```diff
@@ -438,33 +478,27 @@ def calculate_number_of_frames(self) -> int:
                 if n_lines == 0:
                     continue
 
-                try:
-                    n_atoms = int(lines[0].split()[0])
-                except (ValueError, IndexError):
-                    self.logger.error(
-                        (
-                        "Invalid number of atoms in the first line "
-                        f"of file {filename}."
-                        ),
-                        exception=TrajectoryReaderError,
-                    )
+                # Calculate the size of the frame
+                frame_size = self.calculate_frame_size(filename)
```

Comment:
maybe we should add here a Note for our future selfs, that this means it is possible to encounter differen sizes of `StomicSystem`s in different trajectory files given by the user.


```suggestion
                # Calculate the size of the frame
                # NOTE: this allows the user to give input which is not 
                # consisten in the number of atoms per frame in different
                # traj files.
                frame_size = self.calculate_frame_size(filename)
```